lib/cadet/assessments/assessments.ex

```diff
@@ -276,18 +278,24 @@ defmodule Cadet.Assessments do
         Assessment.changeset(%Assessment{}, params)
 
       assessment ->
-        if Timex.after?(assessment.open_at, Timex.now()) do
-          # Delete all existing questions
-          %{id: assessment_id} = assessment
+        %{id: assessment_id} = assessment
 
+        question_count =
+          Question |> where(assessment_id: ^assessment_id) |> Repo.aggregate(:count, :id)
+
+        if Timex.after?(assessment.open_at, Timex.now()) or question_count == new_question_count do
+          # Delete all existing questions
           Question
           |> where(assessment_id: ^assessment_id)
           |> Repo.delete_all()
```

Comment:
I don't think this works for mutation though. What happens if there are existing answers? How will we map the new question to the existing answers. I believe we currently use question assoc when we deal with answers in some parts of the code.

Reply:
Another issue is that... if I'm not wrong, because of the hourly assessment updater task, we are actually deleting and inserting new questions every hour. I can't believe I am seeing question ids in the 3000s in the current SWS version. 

I think this assessment updating logic needs to be reworked.

Reply:
I feel like there has to be a reason why they chose to delete and re insert! let me check with them haha